src/full/Agda/Interaction/Highlighting/LaTeX.hs

```diff
@@ -704,19 +698,10 @@ toLaTeX cc path source hi
       -- Characters which share the same meta info are the same token, so
       -- group them together.
     . groupByFst
-    ))
-
-  -- Characters in different layers belong to different tokens
-  . groupByFst
+    ))) . groupByFst
 
   -- Look up the meta info at each position in the highlighting info.
-  . map (\(pos, (role, char)) -> (role, (IntMap.lookup pos infoMap, char)))
-
-  -- Add position in file to each character.
-  . zip [1..]
-
-  -- Map each character to its role
-  . atomizeLayers . literateTeX (startPos (Just path))
+  . zipWith (curry (\(pos, (role, char)) -> (role, (IntMap.lookup pos infoMap, char)))) [1..] . atomizeLayers . literateTeX (startPos (Just path))
```

Comment:
1.  You can get rid of `curry` here
2. Please keep lines to a reasonable length (preferably 80 characters)
3. Don't remove the comments!!